library/general/disallowanonymous/suite.yaml

```
@@ -14,4 +14,16 @@ tests:
   - name: example-disallowed
     object: samples/no-anonymous-bindings/example_disallowed.yaml
     assertions:
-    - violations: yes
\ No newline at end of file
+    - violations: yes
+  - name: authenticated-allowed-with-parameter-undefined
+    object: samples/no-authenticated/example_disallowed.yaml
+    assertions:
+    - violations: no
+- name: disallow-authenticated
+  template: template.yaml
+  constraint: samples/no-authenticated/constraint.yaml
+  cases:
+    - name: authenticated-disallowed-with-parameter-true
+      object: samples/no-authenticated/example_disallowed.yaml
+      assertions:
+      - violations: yes
```

You might want to consider also asserting the exact expected violation message(s) for higher test coverage.

```suggestion
      - violations: yes
      - message: "{message}"
         violations: 1
```

---

Nice!  Didn't know about this feature.  I've improved the suite test by confirming the exact messages that should be returned.